pkg/authorization/admission/restrictusers/restrictusers.go

```diff
@@ -118,6 +124,39 @@ func (q *restrictUsersAdmission) Admit(a admission.Attributes) (err error) {
 
 	obj, oldObj := a.GetObject(), a.GetOldObject()
 	switch {
+	case gr == rbac.Resource("rolebindings"):
+		rolebinding, ok := obj.(*rbac.RoleBinding)
+		if !ok {
+			return admission.NewForbidden(a,
+				fmt.Errorf("wrong object type for new rolebinding: %T", obj))
+		}
+
+		if len(rolebinding.Subjects) == 0 {
+			return nil
+		}
+		subjects, err = authorizationapi.Convert_rbac_Subjects_To_authorization_Subjects(rolebinding.Subjects)
+		if err != nil {
+			return admission.NewForbidden(a,
+				fmt.Errorf("Failed to convert subjects for rolebinding: %v", err))
+		}
+
+		if oldObj != nil {
+			oldrolebinding, ok := oldObj.(*rbac.RoleBinding)
+			if !ok {
+				return admission.NewForbidden(a,
+					fmt.Errorf("wrong object type for old rolebinding: %T", oldObj))
+			}
+
+			oldSubjects, err = authorizationapi.Convert_rbac_Subjects_To_authorization_Subjects(oldrolebinding.Subjects)
+			if err != nil {
+				return admission.NewForbidden(a,
+					fmt.Errorf("Failed to convert subjects for rolebinding: %v", err))
+			}
+		}
+
+		glog.V(4).Infof("Handling rolebinding %s/%s",
+			rolebinding.Namespace, rolebinding.Name)
+
 	case authorizationapi.IsResourceOrLegacy("rolebindings", gr):
```

Comment:
Why not just enforce on the RBAC resource you proxied to?

Reply:
That is probably what we want to do, though at that point we may need to rewrite most of the logic in this file and the tests to be RBAC specific.  We may want to take a half step and have this working using conversions so that it no longer blocks deployment to online.  Then we can continue to work on this for 3.7 release without rushing.

Reply:
ok I think it's fair, I'll go ahead and actually drop all authorization api related stuff in RBR and tests, and only base on RBAC.

Reply:
@enj, I am still using the conversion functions to convert subjects, as I am not rewriting subjectchecker at this time, we'll probably open a followup issue for that one for now